packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,161 @@
+        collect: (monitor: DragSourceMonitor) => {
+            return {
+                componentDragState: monitor.isDragging() ? ItemDragState.Dragging : ItemDragState.Idle,
+            };
+        },
+        type: listId,
+        canDrag: isEditing,
+    });
+
+    const [copied, setCopied] = useState(false);
+
+    const copyToClipboard = async (text?: string) => {
+        try {
+            const toCopy = text || location.href;
+            await navigator.clipboard.writeText(toCopy);
+        } catch (err) {
+            console.error('Failed to copy: ', err);
+        }
+    };
+
+    const onTooltipClick: MouseEventHandler = (event: MouseEvent) => {
+        event.preventDefault();
+
+        copyToClipboard(getRgbaString(currentColor));
+
+        setCopied(true);
+
+        setTimeout(() => {
+            setCopied(false);
+        }, 2000);
+    };
+
+    const bgColor = `rgba(${backgroundColorRgba}`;
+
+    return (
+        <div
+            id={`row-id-${id}-index-${index}`}
+            style={{
+                width: `${width}px`,
+                left: `${calculateLeftPosition(index, width)}px`,
+                height,
+            }}
+            className={`hover:tw-z-30 row tw-overflow-x-visible tw-overflow-y-hidden tw-pb-8 tw-inline-block
+            `}
+        >
+            <DragHandle index={index} currentColor={currentColor} isEditing={isEditing} onResizeStart={onResizeStart} />
+            <div
+                ref={drag}
+                draggable
+                onDrag={onDrag}
+                onDragEnd={onDragEnd}
+                style={{
+                    height,
+                    width: `${width}px`,
+                }}
+                className={
+                    'tw-group tw-overflow-y-hidden tw-overflow-x-visible tw-top-2 tw-border-[0px] tw-border-white tw-w-full'
+                }
+            >
+                <Tooltip
+                    alignment={TooltipAlignment.Middle}
+                    content={
+                        <div>
+                            <div>{currentColor && currentColor.color && currentColor.color.name}</div>
+                            <div>{getRgbaString(currentColor)}</div>
+                            <a href="#" rel="noop noreferrer" onClick={onTooltipClick} className="tw-opacity-50">
+                                {copied ? 'Copied' : 'Click to copy'}
+                            </a>
+                        </div>
+                    }
+                    flip
+                    heading=""
+                    hoverDelay={200}
+                    position={TooltipPosition.Bottom}
+                    triggerElement={
+                        <div
+                            style={{
+                                backgroundColor: bgColor,
+                                height,
+                                width: `${width}px`,
+                            }}
+                            className={` tw-top-0 tw-left-0
```

Comment:
Same as before, use `first:` and `last:` from tailwind if possible.

Reply:
This seems to be outdated as well: `tw-top-0` doesn't exist in this file anymore.